src/gg.regression.unity.bots/Runtime/Scripts/RGSettings.cs

```diff
@@ -120,9 +121,20 @@ public DebugLogLevel GetLogLevel()
             return logLevel;
         }
         
-        public string GetApiKey() 
+        public string ApiKey
         {
-            return apiKey;
+            get => encodedApiKey != null ? Encoding.UTF8.GetString(Convert.FromBase64String(encodedApiKey)) : null;
+            set => encodedApiKey = Convert.ToBase64String(Encoding.UTF8.GetBytes(value));
+        }
+
+        /**
+         * In some cases, like in the UI registrar, we need to store the API key in a serialized property, rather
+         * than through this class. This method allows us to encode the API key for storage in serialized properties
+         * without the calling location needing to know how we encode and decode.
+         */
+        public static string EncodeApiKeyForSerializedProperties(string decodedApiKey)
```

Comment:
I would leave these changes out of the current PR as was suggested in the reviews.

Encoding/Decoding Base64 does nothing to address the security of the field.  We should only be encoding to Base64 if we have non-serializable char values in the key.

As far as the structure of this code, the get/set for the ApiKey property should use common  Encode/Decode methods that the RGSettingsUIRegistrar would use.  Right now it has code duplication for reading an encoded key

Reply:
Oh ok, I'll revert those commits, I thought the suggestion was that either a secure or non-secure approach was fine, just so people can't directly copy paste. And yes that was a mistake I have fixed in an upcoming commit, but I'll just remove that for now haha